daemon/integration/python/run_many.py

```diff
@@ -0,0 +1,158 @@
+import client_lib as proto
+
+# def StartDaemon():
+# #     call(["$GOPATH/bin/daemon", "-execer_type", "os", "-q_len", "200"])
+#     (["/Users/jbruno/workspace/bin/daemon", "-execer_type", "os", "-q_len", "200"])
+
+
+class TestManyRunRequests(unittest.TestCase):
+    daemonProcess = None
+      
+    def setUp(self):
+        # Note: the following does not work - the process from the pool does not have GOPATH defined so it can't find the binary
+        gopath = os.environ['GOPATH']
+        print("keys {0}".format(os.environ))
+        print("gopath:{0}",gopath)
+        self.daemonProcess = subprocess.Popen(["{0}/bin/daemon".format(gopath), "-execer_type", "os", "-q_len", "200"])
+        time.sleep(1.0)
+        proto.start()
+        
+    def tearDown(self):
+        self.daemonProcess.kill()
+        unittest.TestCase.tearDown(self)
+        
+    sleep_ss_key = "sleep_s_id"
+    ls_ss_key = "ls_s_id"
+    ls_fail_ss_key = "ls_fail_s_id"
+    
+    # test scenario
+    class CmdDef(object):
+        def __init__(self, cmd, snapshot_key, interm_states, final_state):
+            self.cmd = cmd
+            self.snapshot_key = snapshot_key
+            self.interm_states = interm_states
+            self.final_state = final_state
+        
+    sleepDef = CmdDef(cmd=["sh", "./sleep.sh"], snapshot_key="sleep_s_id", interm_states=[proto.ScootStatus.State.PENDING, proto.ScootStatus.State.RUNNING, proto.ScootStatus.State.COMPLETED], final_state=proto.ScootStatus.State.COMPLETED)  # run sleep 1, expect complete or running, final status:complete
+    lsDef = CmdDef(cmd=["sh", "./ls_script.sh"], snapshot_key="ls_s_id", interm_states=[proto.ScootStatus.State.PREPARING, proto.ScootStatus.State.PENDING, proto.ScootStatus.State.RUNNING, proto.ScootStatus.State.COMPLETED], final_state=proto.ScootStatus.State.COMPLETED)  # run ls script 5 times expect complete, running or queued, final status:complete
+    failDef = CmdDef(cmd=["sh", "./ls_script.sh"], snapshot_key="ls_fail_s_id", interm_states=[proto.ScootStatus.State.PENDING, proto.ScootStatus.State.RUNNING, proto.ScootStatus.State.COMPLETED, proto.ScootStatus.State.FAILED], final_state=proto.ScootStatus.State.COMPLETED)  # run ls script 5 times expect complete, running or queued, final status:failed
+
+    # TODO start the server    
+    def test_many(self):
+        tmpdir = tempfile.mkdtemp()
+        rpc_timeout_ns = int(500 * 1e6)
+        
+        runs = {}
+        
+        try:
+        # Make the snapshots.
+            ss_ids = self.make_snapshots()
+        
+            # Run the sleep script
+            id = proto.run(argv=self.sleepDef.cmd, timeout_ns=rpc_timeout_ns, snapshot_id=ss_ids[self.sleepDef.snapshot_key])
+            runs[id] = "sleep"
+            
+            # run ~200 requests, 10 repetitions of 20 successful, 1 error
+            for i in range (0, 10):
+                for j in range(0, 19):
+                    try:
```

Comment:
Having the nested try blocks looks weird.  If the only reason for the larger try block is for the tempdir maybe move the tempdir creation and removal into setUp and tearDown